### PR TITLE
[Android] Implement pairing with code API

### PR DESCRIPTION
**Pull request by joonhaengHeo**
In order to commission on the Android platform, API calls had to be made after parsing the QRCode and Manual Pairing Code.
For this reason, there is a problem of duplicative loading of SetupPayload.jar and so files.
Also, even if there is Controllee in the network, only BLE Commissioning attempts to connect.

To solve this, we implemented the Android BLE code that is not implemented and modified it to use the API in the controller.

### CODE REVIEW

File: src/controller/java/CHIPDeviceController-JNI.cpp [hunk 1]
```diff
@@ -653,6 +653,42 @@ JNI_METHOD(void, pairDeviceWithAddress)
     }
 }
 
+JNI_METHOD(void, pairDeviceWithCode)
```
Comment (yunhanw-google) [inline src/controller/java/CHIPDeviceController-JNI.cpp:656]: Could you add one test for this API in https://github.com/project-chip/connectedhomeip/tree/master/examples/java-matter-controller/java/src/com/matter/controller/commands/pairing

File: src/controller/java/src/chip/devicecontroller/ChipDeviceController.java [hunk 2]
```diff
@@ -175,6 +175,14 @@ public void pairDeviceWithAddress(
         deviceControllerPtr, deviceId, address, port, discriminator, pinCode, csrNonce);
   }
 
+  public void pairDeviceWithCode(
```
Comment (yunhanw-google) [inline src/controller/java/src/chip/devicecontroller/ChipDeviceController.java:178]: Please add doxygen for this public API, thanks

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/ChipClient.kt [hunk 3]
```diff
@@ -61,7 +61,7 @@ object ChipClient {
     if (!this::androidPlatform.isInitialized && context != null) {
       //force ChipDeviceController load jni
       ChipDeviceController.loadJni()
-      androidPlatform = AndroidChipPlatform(AndroidBleManager(), PreferencesKeyValueStoreManager(context), PreferencesConfigurationManager(context), NsdManagerServiceResolver(context), NsdManagerServiceBrowser(context), ChipMdnsCallbackImpl(), DiagnosticDataProviderImpl(context))
+      androidPlatform = AndroidChipPlatform(AndroidBleManager(context), PreferencesKeyValueStoreManager(context), PreferencesConfigurationManager(context), NsdManagerServiceResolver(context), NsdManagerServiceBrowser(context), ChipMdnsCallbackImpl(), DiagnosticDataProviderImpl(context))
```
Comment (yunhanw-google) [inline examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/ChipClient.kt:64]: Is there UI in android chip-tool to exercise this new API? If not, could you add it, thanks

Comment (joonhaengHeo) [reply examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/ChipClient.kt:64]: This code should be pre-executed when AndroidPlatform initializes. I don't think a new UI is needed for this.

File: src/controller/java/src/chip/devicecontroller/ChipDeviceController.java (hunk 2 above)
Comment (joonhaengHeo) [reply src/controller/java/src/chip/devicecontroller/ChipDeviceController.java:187]: I added.

File: src/controller/java/CHIPDeviceController-JNI.cpp (hunk 1 above)
Comment (joonhaengHeo) [reply src/controller/java/CHIPDeviceController-JNI.cpp:656]: I added. (code, code-wifi, code-thread)

**Comment by woody-apple** [comment]
@andy31415 ?

**Review by yunhanw-google — APPROVED** [review]

**Pull request merged**